pilot/pkg/kube/inject/webhook_test.go

```diff
@@ -230,6 +243,290 @@ func TestWebhookInject(t *testing.T) {
 	}
 }
 
-func TestRun(t *testing.T) {
-	// TODO verify run loop handles HTTP serving and config reload properly
+func makeTestData(t testing.TB, skip bool) []byte {
+	t.Helper()
+
+	pod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "test",
+			Annotations: map[string]string{},
+		},
+		Spec: corev1.PodSpec{
+			Volumes:        []corev1.Volume{{Name: "v0"}},
+			InitContainers: []corev1.Container{{Name: "c0"}},
+			Containers:     []corev1.Container{{Name: "c1"}},
+		},
+	}
+
+	if skip {
+		pod.ObjectMeta.Annotations[istioSidecarAnnotationPolicyKey] = "false"
+	}
+
+	raw, err := json.Marshal(&pod)
+	if err != nil {
+		t.Fatalf("Could not create test pod: %v", err)
+	}
+
+	review := v1beta1.AdmissionReview{
+		Request: &v1beta1.AdmissionRequest{
+			Kind: metav1.GroupVersionKind{},
+			Object: runtime.RawExtension{
+				Raw: raw,
+			},
+			Operation: v1beta1.Create,
+		},
+	}
+	reviewJSON, err := json.Marshal(review)
+	if err != nil {
+		t.Fatalf("Failed to create AdmissionReview: %v", err)
+	}
+	return reviewJSON
+}
+
+func createWebhook(t testing.TB, sidecarTemplate string) *Webhook {
+	t.Helper()
+	dir, err := ioutil.TempDir("", "webhook_test")
+	if err != nil {
+		t.Fatalf("TempDir() failed: %v", err)
+	}
+
+	config := &Config{
+		Policy:   InjectionPolicyEnabled,
+		Template: sidecarTemplate,
+	}
+	configBytes, err := yaml.Marshal(config)
+	if err != nil {
+		t.Fatalf("Could not marshal test injection config: %v", err)
+	}
+
+	var (
+		configFile = filepath.Join(dir, "config-file.yaml")
+		meshFile   = filepath.Join(dir, "mesh-file.yaml")
+		certFile   = filepath.Join(dir, "cert-file.yaml")
+		keyFile    = filepath.Join(dir, "key-file.yaml")
+		port       = 0
+	)
+
+	if err := ioutil.WriteFile(configFile, configBytes, 0644); err != nil { // nolint: vetshadow
```

Comment:
Do we clean these files up or rely on normal tmp dir cleanup from the OS?

Reply:
We do now. Just added code to remove the temp directory when the test completes or errors out.